### PR TITLE
UT-3723 enable code coverage when testing on yamato

### PR DESCRIPTION
**Pull request by vkovec**
Coverage report can be found under artifacts after it runs:
image

### CODE REVIEW

File: .yamato/yamato.yml [hunk 1]
```diff
@@ -158,7 +158,13 @@ test_{{ platform.name }}_{{ editor.version }}_{{ backend.name }}:
 {% endif %}
   commands:
     - npm install -g upm-ci-utils@stable --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
+# Code coverage is only available in Unity 2019.3 and higher
+{% if editor.version == 2018.4 %}
     - upm-ci package test --backend {{ backend.name }} --unity-version {{ editor.version }} --package-path com.autodesk.fbx
+{% else %}
+    - upm-ci package test --backend {{ backend.name }} --unity-version {{ editor.version }} --package-path com.autodesk.fbx --enable-code-coverage --code-coverage-options 'generateHtmlReport;assemblyFilters:+Autodesk.Fbx'
+    - python tests/Yamato/check_coverage_percent.py upm-ci~/test-results/ 94.7
```
Comment (benoithudson) [inline .yamato/yamato.yml:166]: Can you make this 94.7 a variable? (Up at the top of the file before the `---`)

File: tests/Yamato/check_coverage_percent.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from __future__ import print_function
```
Comment (benoithudson) [inline tests/Yamato/check_coverage_percent.py:1]: Should we have a copyright header?

File: tests/Yamato/check_coverage_percent.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    # Couldn't find the results in the file.
+    print("Couldn't find Linecoverage in summary file")
+    sys.exit(1)
+
+
+def main():
+    root_dir = sys.argv[1]
+    min_percent = float(sys.argv[2])
+    if min_percent > 0:
+        # This allows us to set 0% coverage on 2018.4
```
Comment (benoithudson) [inline tests/Yamato/check_coverage_percent.py:55]: I don't understand this comment, but if it's just copied code then that's fine.

File: tests/Yamato/check_coverage_percent.py (hunk 2 above)
Comment (benoithudson) [reply tests/Yamato/check_coverage_percent.py:1]: Also please mention where you got this, so we can remember to look for newer versions.

**Review by benoithudson — CHANGES_REQUESTED** [review]
Minor changes noted.

I'm thinking a changelog entry saying we're now using Unity code coverage would be good for our own memory, and would advertise the package a bit.

So cool! Much better than the older code.

File: tests/Yamato/check_coverage_percent.py (hunk 3 above)
Comment (vkovec) [reply tests/Yamato/check_coverage_percent.py:55]: yes it's copied

File: tests/Yamato/check_coverage_percent.py (hunk 2 above)
Comment (vkovec) [reply tests/Yamato/check_coverage_percent.py:1]: not sure, the original doesn't have one.

Comment (benoithudson) [reply tests/Yamato/check_coverage_percent.py:3]: The package license.md has all we need I think. Thanks for the added line!

**Review by benoithudson — APPROVED** [review]

**Pull request merged**